src/matsim_io/writers.py

```diff
@@ -153,6 +154,9 @@ def add_leg_with_route(
         self._write(f'<leg mode="{mode}"')
         if departure_time is not None:
             self._write(f' dep_time="{self.time(departure_time)}"')
+        if mat_sim_routing:
```

Comment:
Maybe change the order og this statement and the next. self._write("/>\n") has to execute no matter what so maybe do that and then do if matsimrouting return